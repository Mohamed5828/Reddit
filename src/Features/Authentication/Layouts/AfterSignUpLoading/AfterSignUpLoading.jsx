/* eslint-disable no-unused-vars */
/* eslint-disable react-hooks/exhaustive-deps */

import LoadingSpinner from "Features/Authentication/Components/LoadingSpinner/LoadingSpinner";

import {
  AuthContainer,
  Privacy,
  AuthParagraph,
  AuthHeader,
  RedditIcon,
  Upper,
  Skip,
} from "./AfterSignUpLoading.styled";

/**
 *  AfterSignUpLoading component that is used in AfterSignUp Component
 * @returns {React.Component}  AfterSignUpLoading component that is used in AfterSignUp Component
 */

const AfterSignUpLoading = () => {
  return (
    <>
      {
        <AuthContainer>
            <AuthHeader>We're happy you're here!</AuthHeader>
            <AuthParagraph>Your home feed is being created now...</AuthParagraph>
          <LoadingSpinner />
        </AuthContainer>
      }
    </>
  );
};

export default AfterSignUpLoading;
