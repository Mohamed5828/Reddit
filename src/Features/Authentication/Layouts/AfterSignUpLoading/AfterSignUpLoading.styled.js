import styled from "styled-components";
import { createGlobalStyle } from "styled-components";

const fontFamily = "'IBM Plex Sans', sans-serif";

export const AuthContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin: 0px auto 0;
  position: relative;
  width: 100%;
  background-image: url(https://www.redditstatic.com/desktop2x/img/confetti_background.png);
  height: 100%;
  min-width: 102%;
  position: absolute;
  top: 0;
  left: -4px;
  justify-content: center;
  align-items: center;
  border-radius: 10px;

  .spinner-border {
    color: white;
  }
`;

export const AuthHeader = styled.h2`
  font-family: ${fontFamily};
  line-height: 24px;
  max-width: 280px;
  text-transform: none;
  display: flex;
  justify-content: center;
  font-weight: bold;
  font-size: 22px;
  margin: 10px 0;
  color: white;
`;
export const AuthParagraph = styled.div`
  color: white;
  font-family: Noto Sans, sans-serif;
  font-size: ${(p) => (p.identity ? "14px" : "12px")};
  font-weight: ${(p) => (p.identity ? "550" : "400")};
  line-height: 16px;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 14px;
  margin: 10px 0;
`;

export const Privacy = styled.span`
  color: ${({ theme }) => theme.color.link};
`;

export const RedditIcon = styled.div`
  display: flex;
  justify-content: center;
  width: 100%;

  svg {
    transform: translateX(18px);
  }
`;

export const Upper = styled.div`
  display: flex;
  justify-content: center;

  span {
    cursor: pointer;
    font-size: 16px;
    font-weight: 600;
  }
`;

export const Skip = styled.span``;

export const GlobalStyles = createGlobalStyle`
.modal-body {
    position: relative;
    flex: 1 1 auto;
    padding: 0;
}`;
