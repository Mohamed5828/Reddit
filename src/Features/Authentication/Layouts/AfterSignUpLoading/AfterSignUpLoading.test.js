import { shallow } from "enzyme";
import AfterSignUpLoading from "./AfterSignUpLoading";

describe("AfterSignUpLoading Component", () => {
  it("this is a test for AfterSignUpLoading Component", () => {
    expect(shallow(<AfterSignUpLoading />)).toMatchSnapshot();
  });
});
