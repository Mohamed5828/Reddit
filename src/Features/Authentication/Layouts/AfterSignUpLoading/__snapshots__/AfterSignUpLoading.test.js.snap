// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AfterSignUpLoading Component this is a test for AfterSignUpLoading Component 1`] = `
<Fragment>
  <styled.div>
    <styled.h2>
      We're happy you're here!
    </styled.h2>
    <styled.div>
      Your home feed is being created now...
    </styled.div>
    <LoadingSpinner />
  </styled.div>
</Fragment>
`;
